lib/semantics/symbol.cc

```diff
@@ -530,6 +531,33 @@ static void DumpUniqueName(std::ostream &os, const Scope &scope) {
   }
 }
 
+// Dump flag list for a symbol in Unparse
+static void DumpFlagsForUnparse(std::ostream &os, const Symbol::Flags &flags) {
+  static std::unordered_map<Symbol::Flag, const std::string> flagMap{
+      {Symbol::Flag::Implicit, "implicit"},
+      {Symbol::Flag::LocalityLocal, "local"},
+      {Symbol::Flag::LocalityLocalInit, "local_init"},
+      {Symbol::Flag::LocalityShared, "shared"}};
+  std::size_t seen{0};
+
+  flags.IterateOverMembers([&](Symbol::Flag flag) {
```

Comment:
I think I would reverse the logic here and iterate through `flagMap` instead of `flags`, `IterateOverMembers` looks like it is doing a copy of the set to perform iterations, plus the lambda looks more complex to me than:
```
for (const auto& pair: flagMap) {
  if (flags.test(pair.first)) {
       if (seen++ > 0) {
          os << ", ";
        } else {
          os << " (";
        }
        os << pair.second;
  }
}
```

If you go for this you should probably also make `flagMap` a `std::map` instead of an `std::unordered_map` to guarantee the output order.

Reply:
Thanks, Jean, I will change the logic here.